From 2bd77483e8a7caa0c14762f2cc37e2a0d94c950b Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 13 Apr 2018 14:55:33 +0300
Subject: [PATCH 2822/5242] MLK-16819-3: dts: Add HDMI CEC property to imx8mq
 dtb

commit  25d2b4611b50e358ed12e6e34897e720ee29f749 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add imx8 HDMI CEC property to imx8mq dtb.
Change HDMI HDP Controller memory map size.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-drm.dts      |    6 +++++-
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |   10 ++++++++--
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts
index 3d4919b..d2df468 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts
@@ -68,7 +68,7 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 		compatible = "fsl,imx8mq-hdmi";
-		reg = <0x0 0x32c00000 0x0 0x100000>,	/* HDP registers */
+		reg = <0x0 0x32c00000 0x0 0x33800>,	/* HDP registers */
 				<0x0 0x32e40000 0x0 0x40000>;   /* HDP SEC register */
 		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 		power-domains = <&disp_pd>;
@@ -81,6 +81,10 @@
 	};
 };
 
+&hdmi_cec {
+	status = "okay";
+};
+
 &dcss {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 73bb17b..fb233f7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -1095,8 +1095,8 @@
 
 	hdmi: hdmi@32c00000 {
 		compatible = "fsl,imx8mq-hdmi";
-		reg = <0x0 0x32c00000 0x0 0x100000>,	/* HDMI registers */
-				<0x0 0x32e40000 0x0 0x40000>;   /* HDMI SEC register */
+		reg = <0x0 0x32c00000 0x0 0x33800>,	/* HDP registers */
+				<0x0 0x32e40000 0x0 0x40000>;   /* HDP SEC register */
 		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 		video-mode = <16>;	/* <default_mode: #16>
 					 * #16: 1920x1080p@60HZ 16:9
@@ -1107,6 +1107,12 @@
 		status = "disabled";
 	};
 
+	hdmi_cec: hdmi_cec@32c33800 {
+		compatible = "fsl,imx8-hdp-cec";
+		reg = <0x0 0x32c33800 0x0 0x200>;
+		status = "disabled";
+	};
+
 	pcie0: pcie@0x33800000 {
 		compatible = "fsl,imx8mq-pcie", "snps,dw-pcie";
 		reg = <0x0 0x33800000 0x0 0x400000>, <0x0 0x1ff00000 0x0 0x80000>;
-- 
1.7.9.5

